molgenis-data-excel/src/main/java/org/molgenis/data/excel/xlsx/XlsxWriter.java

```diff
@@ -0,0 +1,161 @@
+    } catch (RuntimeException e) {
+      throw new XlsxWriterException(e);
+    }
+  }
+
+  public void createSheet(String name, List<Object> headers) {
+    try {
+      if (name.length() <= MAXIMUM_SHEET_LENGTH) {
+        Sheet sheet = workbook.getSheet(name);
+        if (sheet == null) {
+          sheet = workbook.createSheet(name);
+          internalWriteRow(headers, sheet, 0);
+        }
+      } else {
+        throw new MaximumSheetNameLengthExceededException(name);
+      }
+    } catch (MaximumSheetNameLengthExceededException e) {
+      throw e;
+    } catch (RuntimeException e) {
+      throw new XlsxWriterException(e);
+    }
+  }
+
+  /**
+   * @param row List of Objects, allowed Object classes: Boolean, LocalDate, Instant, Double,
+   *     Integer, Long, String
+   * @param sheetName
+   */
+  public void writeRow(List<Object> row, String sheetName) {
+    try {
+      this.writeRows(Stream.of(row), sheetName);
+    } catch (RuntimeException e) {
+      throw new XlsxWriterException(e);
+    }
+  }
+
+  /**
+   * @param rows List of Lists of Objects, allowed Object classes: Boolean, LocalDate, Instant,
+   *     Double, Integer, Long, String
+   * @param sheetName
+   */
+  public void writeRows(List<List<Object>> rows, String sheetName) {
+    try {
+      this.writeRows(rows.stream(), sheetName);
+    } catch (RuntimeException e) {
+      throw new XlsxWriterException(e);
+    }
+  }
+
+  /**
+   * @param rows Stream of Lists of Objects, allowed Object classes: Boolean, LocalDate, Instant,
+   *     Double, Integer, Long, String
+   * @param sheetName
+   */
+  public void writeRows(Stream<List<Object>> rows, String sheetName) {
+    try {
+      Sheet sheet = workbook.getSheet(sheetName);
+      rows.forEach(
+          row -> {
+            internalWriteRow(row, sheet, sheet.getLastRowNum() + 1);
+          });
+    } catch (RuntimeException e) {
+      throw new XlsxWriterException(e);
+    }
+  }
+
+  public void close() throws IOException {
+    try {
+      workbook.write(Files.newOutputStream(target));
+    } catch (RuntimeException e) {
+      throw new XlsxWriterException(e);
+    } finally {
+      workbook.close();
+    }
+  }
+
+  private void internalWriteRow(List<Object> values, Sheet sheet, int rowNr) {
+    final Row row = sheet.createRow(rowNr);
+    AtomicInteger counter = new AtomicInteger(0);
+    values
+        .stream()
+        .forEach(
+            record -> {
+              int index = counter.getAndIncrement();
+              if (record != null) {
+                Cell cell = row.createCell(index);
+                setCellValue(cell, record);
+              }
+            });
+  }
+
+  protected void setCellValue(Cell cell, Object value) {
+    if (value instanceof Boolean) {
+      cell.setCellValue(toBoolean(value));
+    } else if (value instanceof LocalDate) {
+      Instant instant = toLocalDate(value).atStartOfDay(timeZone.toZoneId()).toInstant();
```

Comment:
Potential performance bottleneck to create a ```SimpleDateFormat``` for every cell. How about setting a field with a default format in the constructor and add setters so the user can select a different format (e.g. without timezone)?

Reply:
Sorry for stating the obvious but here's your standard warning not to share SimpleDateFormats between threads or you'll get very interesting results. It's not thread safe.

Reply:
Jup, but I don't see any threading issues when sharing a format within an instance of XlsxWriter.